src/sql/workbench/contrib/notebook/common/utils.ts

```diff
@@ -0,0 +1,14 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export const invalidRelativePathRegex = /^\.\.+(?=\.\.)/g;
+
+export function replaceInvalidLinkPath(href: string): string {
+	// Get first slash of link and use that create relative path format (..\) string
+	// and then concatenate relative path format string to rest of href path after slash
+	let slashIndex = href.indexOf('\\');
```

Comment:
For something like this I think you should 

1. Make it defensive against paths that don't necessarily start with .... such that it returns the original string
2. Indicate that it's meant for windows only. Maybe even make a check here that just returns the original string if it's not windows?
3. Be clearly documented regardless about what it does/doesn't support so someone looking at this later is clear on what it's supposed to be accomplishing

Reply:
I guess one question I have is if we also have the checks here for isWindows and string.startswith then what would be the point for having them in the conditions that call this function?

Reply:
That's a good question - and in general I think it would be fine (and easier to understand) to get rid of those checks if the function is able to handle all cases regardless.

Reply:
I think the only reason I would not go down that route is that we would have to take in all the cases from notebookMarkdown and htmlMarkdownConvertor (since the case is in an else if condition). I will keep it for how it is now since it is defensive against all paths, as it might be easier to call the function in the future depending on what the cases look like via that entry point.